### DISCUSSION TITLE
OVT at edge server delays video

### DISCUSSION THREAD

**Discussion by web3omega** [OvenMediaLabs/OvenMediaEngine#1284] [Help]
I have a central origin server connected to several edge servers spread across the world.
The video streams from origin to edge are using OVT.

I use WEBRTC and LLHLS on the edge server to play video realtime.

**Issue:**
Over time the video on the edge server is played at a slower speed, but also delayed. Its sometimes delayed by several minutes! I also noticed the origin server is having increased memory allocation (More than 2GB instead of nominal 320MB) on the ovenmediaengine process. Why is this drifting?

**Question**: Is this a bug? Or Is there a way to force the delay of the OVT stream to be max x seconds OR set the max size to buffer?

My workaround: I add a memory limit above 2 GB, and let ovenmediaengine crash + restart. This forces the stream to restart and the problem is gone until it occurs at a later moment again.

[origin_server.txt](https://github.com/AirenSoft/OvenMediaEngine/files/11860516/origin_server.txt)
[edge_server.txt](https://github.com/AirenSoft/OvenMediaEngine/files/11860517/edge_server.txt)

**Comment by getroot** [comment]
Based on the information given, it seems that the network speed between Origin and Edge is slow. It seems that the data is constantly piling up in the queue because the actual data being sent is less than the amount of data that the origin needs to send.

**Reply by web3omega** [reply]
Yes this makes sense, because this edge server is on the other side of the world than the origin server. But i have 1gbit uplink on the origin and a 10gbit on the edge.

Regardless of the network speed which i can potentially improve, is there a way to avoid piling up the queue?

**Reply by getroot** [reply]
Currently, Origin and Edge are assumed to have sufficient speed because they operate on a managed network. Therefore, the function to drop packets when the network speed between origin and edge is continuously slow and the queue continues to grow is not currently implemented. If the origin drops packets, the playback quality from the edge is very poor, and it is very difficult to find the cause of this problem. I will think more about this. If you have any good ideas please tell me.

The ability to automatically delete lazy streams (with ever-growing queues) is provided. But in this case, the origin's stream will be deleted, so I don't think it will fit your case. https://github.com/AirenSoft/OvenMediaEngine/discussions/1221
